protocol/standard-validator.md

```diff
@@ -0,0 +1,153 @@
+
+In order to clarify the cause of these `expectedErrors` it is necessary to clarify the jobs the
+StandardValidator is currently doing:
+
+1. Enforcing the standard configuration.
+2. Checking that values are 'as expected', even for chains which do not satisfy the standard configuration.
+
+To clarify the latter, chains may have a different L1ProxyAdminOwner (ie.
+[Unichain](https://github.com/ethereum-optimism/superchain-registry/blob/0831c2509152b457d865634616925ca6240b219e/superchain/configs/mainnet/unichain.toml#L45)),
+or a different Challenger (ie.
+[Base](https://github.com/ethereum-optimism/superchain-registry/blob/0831c2509152b457d865634616925ca6240b219e/superchain/configs/mainnet/base.toml#L47)).
+In such cases the StandardValidator currently returns an error string (ie. `PROXYA-10` or `PDDG-120`),
+  the current practice is to recognize that this error is expected, but does not allow the StandardValidator
+  to make assertions on the expected value of these roles.
+
+## Proposed Solution
+
+The proposed solution is broken up in to work that can be done prior to upgrade 16, and work
+that will come afterwards.
+
+### Pre-upgrade 16 work
+
+#### 1. OPCM integration:
+
+The `StandardValidator` should:
+- be deployed by [DeployImplementations](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/scripts/deploy/DeployImplementations2.s.sol#L39)
+- included on the OPCM as an [immutable variable](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/OPContractsManager.sol#L1785-L1788)
+- listed in the [standard-versions](https://github.com/ethereum-optimism/superchain-registry/blob/0831c2509152b457d865634616925ca6240b219e/validation/standard/standard-versions-mainnet.toml#L1) toml files
+
+#### 2. Improved validations:
+
+The `StandardValidator` should correctly handle certain allowable/expected deviations.
+
+A non-exhaustive list includes:
+
+- `absolutePrestate`: this value varies for each chain. The [current design](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/StandardValidator.sol#L61) accepts this value as an input, so no change is needed.
+- Various roles: the default roles should be hardcoded into the StandardValidator, but calls to
+  `validate()` should optionally override them.
+  - note: The
+    [guardian](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/StandardValidator.sol#L138)
+    is not currently checked, and the paused state should not be checked.
+- The Permissionless Dispute Game: [should be
+  optional](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/StandardValidator.sol#L368).
```

Comment:
Note that with U16 we also add a super dispute game and permissioned super dispute game, so 4 game types now, and eventually we may add asterisc (5th game type)

Reply:
Two acceptable states for `gameImpls`
1. both super games set AND no other games
2. no super games AND (permissioned only OR both non-super games set)

Then verify that `respectedGameType` is one of the games in gameImpls